diff -Nur binutils-2.19.1/binutils/readelf.c binutils-2.19.1-irix/binutils/readelf.c
--- binutils-2.19.1/binutils/readelf.c	2008-09-17 09:00:44.000000000 +0000
+++ binutils-2.19.1-irix/binutils/readelf.c	2025-04-26 05:35:59.793977520 +0000
@@ -165,7 +165,7 @@
 #include "safe-ctype.h"
 
 char *program_name = "readelf";
-int do_wide;
+extern int do_wide;
 static long archive_file_offset;
 static unsigned long archive_file_size;
 static unsigned long dynamic_addr;
diff -Nur binutils-2.19.1/gas/dwarf2dbg.h binutils-2.19.1-irix/gas/dwarf2dbg.h
--- binutils-2.19.1/gas/dwarf2dbg.h	2007-11-19 19:37:18.000000000 +0000
+++ binutils-2.19.1-irix/gas/dwarf2dbg.h	2025-04-26 05:35:59.802977704 +0000
@@ -81,12 +81,12 @@
 
 /* True when we've seen a .loc directive recently.  Used to avoid
    doing work when there's nothing to do.  */
-bfd_boolean dwarf2_loc_directive_seen;
+/*extern bfd_boolean dwarf2_loc_directive_seen;*/
 
 /* True when we're supposed to set the basic block mark whenever a label
    is seen.  Unless the target is doing Something Weird, just call 
    dwarf2_emit_label.  */
-bfd_boolean dwarf2_loc_mark_labels;
+/*extern bfd_boolean dwarf2_loc_mark_labels;*/
 
 extern void dwarf2_finish (void);
 
diff -Nur binutils-2.19.1/intl/dcigettext.c binutils-2.19.1-irix/intl/dcigettext.c
--- binutils-2.19.1/intl/dcigettext.c	2006-05-22 15:30:13.000000000 +0000
+++ binutils-2.19.1-irix/intl/dcigettext.c	2025-04-26 06:19:27.576598131 +0000
@@ -148,7 +148,7 @@
 char *getwd ();
 #  define getcwd(buf, max) getwd (buf)
 # else
-char *getcwd ();
+/*char *getcwd ();*/
 # endif
 # ifndef HAVE_STPCPY
 static char *stpcpy PARAMS ((char *dest, const char *src));
diff -Nur binutils-2.19.1/irix-conf.sh binutils-2.19.1-irix/irix-conf.sh
--- binutils-2.19.1/irix-conf.sh	1970-01-01 00:00:00.000000000 +0000
+++ binutils-2.19.1-irix/irix-conf.sh	2025-04-26 05:35:59.802977704 +0000
@@ -0,0 +1,5 @@
+Cross:
+../configure --prefix=/opt/cross-irix5-new --target=mips-sgi-irix5 --without-nls --disable-werror --with-sysroot=/opt/cross-irix5-new/sysroot --disable-pgo-build --enable-lto=no --disable-plugins
+
+Native:
+../configure --prefix=/opt/dev-irix5 --target=mips-sgi-irix5 --host=mips-sgi-irix5 -build=x86_64-linux-gnu --without-nls --disable-werror
