diff -Nur gcc-15.1.0/gcc/config/mips/iris6.h gcc-15.1.0-irix/gcc/config/mips/iris6.h
--- gcc-15.1.0/gcc/config/mips/iris6.h	2025-06-25 05:44:27.447031094 +0000
+++ gcc-15.1.0-irix/gcc/config/mips/iris6.h	2025-06-25 05:43:26.069346826 +0000
@@ -25,13 +25,13 @@
 
 /* Default to -mabi=n32 and -mips3.  */
 #undef MULTILIB_DEFAULTS
-#define MULTILIB_DEFAULTS { "mabi=32" }
+#define MULTILIB_DEFAULTS { "mabi=n32" }
 
 /* Force the default ABI onto the command line in order to make the specs
    easier to write.  Default to the mips2 ISA for the O32 ABI.  */
 #undef DRIVER_SELF_SPECS
 #define DRIVER_SELF_SPECS 			\
-  "%{!mabi=*: -mabi=32}", 			\
+  "%{!mabi=*: -mabi=n32}", 			\
   "%{mabi=32: %{!mips*: %{!march*: -mips2}}}", 	\
   "%{mabi=n32: %{!mips*: %{!march*: -mips3}}}",	\
   "%{mabi=64: %{!mips*: %{!march*: -mips4}}}",	\
